src/common/types/vector.cpp

```diff
@@ -764,7 +764,7 @@ void Vector::Orrify(idx_t count, VectorData &data) {
 		break;
 	default:
 		Normalify(count);
-		data.sel = FlatVector::IncrementalSelectionVector(count, data.owned_sel);
+		data.sel = FlatVector::IncrementalSelectionVector();
```

Comment:
Should we perhaps modify `FlatVector:: IncrementalSelectionVector` to ignore the count and always return `FlatVector::IncrementalSelectionVector`?

Reply:
I think this gets back to the whole issue of `const`ness. From that perspective, `FlatVector:: IncrementalSelectionVector()` should return a `const SelectionVector*`. This then implies that `VectorData.sel` should also be a `const SelectionVector*`. It also suggests that `FlatVector:: IncrementalSelectionVector(count,...)` should _always_ populate the passed `SelectionVector` and return a writable `SelectionVector*`.

This is a bigger change that I'm happy to do, but its a lot more work than fixing the failures. Shall we break it up into two parts?